services/localizely/localizely-base.js

```diff
@@ -0,0 +1,64 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const { BaseJsonService } = require('..')
+
+const keywords = [
+  'l10n',
+  'i18n',
+  'localization',
+  'internationalization',
+  'translation',
+  'translations',
+]
+
+const documentation = `
+  <p>
+    The read-only API token from the Localizely account is required to fetch necessary data.
+    <br/>
+    You can find more details regarding API tokens under <a href="https://app.localizely.com/account" target="_blank">My profile</a> page.
+  </p>
+  `
+
+const schema = Joi.object({
+  strings: Joi.number().required(),
+  reviewedProgress: Joi.number().required(),
+  languages: Joi.array()
+    .items(
+      Joi.object({
+        langCode: Joi.string().required(),
+        langName: Joi.string().required(),
+        strings: Joi.number().required(),
+        reviewed: Joi.number().required(),
+        reviewedProgress: Joi.number().required(),
+      })
+    )
+    .required(),
+}).required()
+
+const queryParamSchema = Joi.object({ token: Joi.string() }).required()
+
+class BaseLocalizelyService extends BaseJsonService {
+  async fetch({ projectId, branch, apiToken }) {
+    return this._requestJson({
+      schema,
+      url: `https://api.localizely.com/v1/projects/${projectId}/status`,
+      options: {
+        qs: { branch },
+        headers: { Accept: 'application/json', 'X-Api-Token': apiToken },
+      },
+      errorMessages: {
+        400: 'bad request',
+        401: 'unauthorized',
+        403: 'forbidden',
+      },
```

Comment:
The custom error message capability exists to allow service classes provide contextual error messages to the end user. The bad request scenario really shouldn't be possible. If there's inputs required by the upstream service, then we need to ensure our badge route, so I wouldn't expect the 401 case to be a real possibility either. 

If there's a desire to provide a custom error message for the 403 case, I'd suggest something that'll shed more light (`invalid API token` or `not authorized for project`, etc.)

Reply:
Suggestion applied within [dcd1d](https://github.com/badges/shields/pull/5472/commits/dcd1d001676e386345e47b46782f60fe5b92339d) commit.